### PR TITLE
Datadog RUM events

### PR DESCRIPTION
**Pull request by wrighbr**
## Hey, I just made a Pull Request!
Adding greater observability with Datadog RUM  events 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Fox32** [comment]
A while back we decided to remove provider specific instrumentation implementations from Backstage, for example Prometheus in #3575. Instead we left it up to the integrators of Backstage on how he wants to instrument his application. He can do that by adding the required tools in his application. Adding direct support for Datadog feels very tool specific. Luckily your changes are primarily inside the example app, so you should be able to integrate Datadog without changes to Backstage. However, I'm not sure about the changes inside the CLI.

I think these changes are better suited as a guide in the `contrib/` folder.

Two small notes in case this is still desired to be added to Backstage: 
* Adding new configuration options should be reflected inside the config schema.
* Beside changing the example-app, you should also apply the changes to the create-app template.

**Comment by wrighbr** [comment]
@Fox32 The idea behind this PR is the same as the Google Analytics Support PR so we can get Analytics but with datadog instead of Google
https://github.com/backstage/backstage/pull/2934

File: .changeset/tiny-ears-love.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': major
```
Comment (freben) [inline .changeset/tiny-ears-love.md:2]: This should be patch if nonbreaking, or minor if breaking

**Comment by wrighbr** [comment]
> Thanks! Looks alright although it seems like a bunch of extra changes got included by mistake, with some files being mistakenly deleted.

@Rugvip I've reverted / removed the files that aren't required for this PR.

**Review by freben — APPROVED** [review]
Yep I think this makes sense at this point, until we build better facilities for getting this kind of integration in there

File: docs/integrations/datadog-rum/installation.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+id: installation
+title: Datadog RUM Installation
+sidebar_label: Installation
```
Comment (timbonicus) [inline docs/integrations/datadog-rum/installation.md:4]: Should this be linked in `sidebars.json`? It won't show up in the navigation otherwise.

Comment (freben) [reply docs/integrations/datadog-rum/installation.md:4]: Would you do the honors?

**Comment by regicsolutions** [comment]
@wrighbr does the https://www.npmjs.com/package/@datadog/browser-rum package need to be installed in the backend for this to work? i tried adding a JS RUM application and added this to my app-config and am not getting any session data come in...

```
   app:
      title: My App
      baseUrl: https://mybackstage-instance.com
      datadogRum:
        clientToken: 11a11111-111a-1a1a-11a1-aa1aaaaa11a1
        applicationId: abc11111a1aa1a11aaa1111a11a1111a111
        env: dev
```

**Comment by wrighbr** [comment]
The datadog package is not required. The site is set to datadoghq.com if this is not correct you will need to change it

**Comment by regicsolutions** [comment]
Thanks for the reply @wrighbr i confirmed the site is `datadoghq.com` in datadog when i add the application in Real User Monitoring, i also called it backstage given it looks like this is hard coded, is there anything else you did aside from adding what i mentioned above in the app-config.yaml to get this working?

**Comment by regicsolutions** [comment]
@wrighbr i also added the script to /packages/app/public/index.html and still not getting anything in datadog.

@freben @timbonicus Datadog has another option using the npm package: https://www.npmjs.com/package/@datadog/browser-rum 

Provides a code sample to initialize it, where would be the best place to initialize this?

```
import { datadogRum } from '@datadog/browser-rum';

datadogRum.init({
    applicationId: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXX',
    clientToken: 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
    site: 'datadoghq.com',
    service:'backstage',
    sampleRate: 100,
    trackInteractions: true
});
```

**Comment by regicsolutions** [comment]
@freben @timbonicus friendly ping 🙂

**Comment by timbonicus** [comment]
@regicsolutions I guess you would add this in `packages/app/src/index.tsx`?

**Comment by wrighbr** [comment]
@regicsolutions  I don't know if you are still having a problem with this. 
But we have a separated frontend and backend and we noticed that our production wasn't showing up in datadog. 
so we added the below when building backstage which seems to solve the problem.

```
  export APP_CONFIG_app_datadogRum_env="$ENV"
  yarn workspace backend build
  yarn workspace app build
```

**Pull request merged**